src/cockpit/packages.py

```diff
@@ -257,26 +241,44 @@ def get_content_security_policy(self, origin):
 
         return ' '.join(f'{k} {v};' for k, v in policy.items()) + ' block-all-mixed-content'
 
-    def serve_file(self, path, channel):
-        filename = self.negotiate_file(path, channel.headers)
-
-        if filename:
-            with (self.path / filename).open('rb') as file:
-                data = file.read()
-        # HACK: if a translation file is missing, just return empty
-        # content. This saves a whole lot of 404s in the developer
-        # console when trying to fetch po.js for English, for example.
-        elif path.endswith('po.js'):
-            data = b''
-            filename = 'po.js'
+    def find_po_js(self, channel):
+        accept_language = channel.headers.get('Accept-Language', '')
+        locales = parse_accept_language(accept_language)
+
+        # First check the locales that the user sent
+        for locale in locales:
+            translation = self.translations.get(locale)
+            if translation is not None:
+                return translation
+
+        # Next, check the language-only versions of variant-specified locales
+        for locale in locales:
+            language, _, region = locale.partition('-')
+            if not region:
+                continue
+            translation = self.translations.get(language)
+            if translation:
+                return translation
+
+        # If nothing else worked, we always have English
+        return self.translations['en']
+
+    def find_file(self, path, channel):
+        if path == 'po.js':
+            # We do locale-dependent lookup, only for /po.js
+            return self.find_po_js(channel)
         else:
-            logging.error("filename=%s, path=%s", filename, path)
+            return self.files.get(path)
+
+    def serve_file(self, path, channel):
+        found = self.find_file(path, channel)
+        if found is None:
+            logging.error("path=%s", path)
             logger.debug('path %s not in %s', path, self.files)
```

Comment:
Redundant logging

Reply:
Dunno why this was an error.  404s don't belong in the journal.